### PR TITLE
Fix RGBWW colors

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

Fix RGBWW colors

Round values and convert percentage to values correctly.

## Type of change

- [ ] Dependency upgrade
- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by emontnemery — APPROVED** [review]

**Review by emontnemery — APPROVED** [review]

File: src/data/light.ts [hunk 1]
```diff
@@ -56,7 +56,13 @@ export const lightSupportsDimming = (entity: LightEntity) => {
   );
 };
 
-export const getLightRgbColor = (entity: LightEntity): number[] | undefined =>
+export const getLightRgbColor = (
+  entity: LightEntity
+): [number, number, number] | undefined => entity.attributes.rgb_color;
+
+export const getLightRawRgbColor = (
```
Comment (balloob) [inline src/data/light.ts:63]: What's raw about this? Maybe name it `getLightCurrentModeRgbColor` ?

File: src/data/light.ts [hunk 2]
```diff
@@ -56,7 +56,13 @@ export const lightSupportsDimming = (entity: LightEntity) => {
   );
 };
 
-export const getLightRgbColor = (entity: LightEntity): number[] | undefined =>
+export const getLightRgbColor = (
```
Comment (balloob) [inline src/data/light.ts:59]: Is this function still needed?

Comment (bramkragten) [reply src/data/light.ts:59]: Yeah, it is used for state badge

File: src/data/light.ts (hunk 1 above)
Comment (bramkragten) [reply src/data/light.ts:63]: We have rgb, that is a representative for the current color, and we have the raw rgb color, without the white channel values

File: src/data/light.ts (hunk 2 above)
Comment (balloob) [reply src/data/light.ts:59]: I meant that since we use typescript, we can just inline it.

File: src/data/light.ts (hunk 1 above)
Comment (balloob) [reply src/data/light.ts:63]: I am not completely following. You're saying that `entity.attributes.rgbww_color` does not contain a white value ?

Comment (bramkragten) [reply src/data/light.ts:63]: Yes it does, but the rgb of rgbww_color is not the same as rgb_color as that last one has the ww mixed in the rgb

**Pull request merged**